### ISSUE TITLE
Warning about unmapped not existing property

### ISSUE THREAD

**Issue by hejcz** [mapstruct/mapstruct#2635]
- [x] Is this an issue (and hence not a question)? 

I have a problem with mapping jdk 17 records. Here is a repo reproducing problem: https://github.com/hejcz/mapstruct-bug 

Once I run 

```
./gradlew clean javaCompile 
```

Warning below is raised:

```
> Task :compileJava FAILED
> Task :compileJava
/home/jrubin/IdeaProjects/mapstruct-test/src/main/java/io/github/hejcz/DtoMapper.java:31: warning: Unmapped target property: "hes".
    WishList mapWishlist(WishList2 wishlist);
             ^
1 warning
```

Even though as far as I understand all fields are mapped and there is no field "hes". When I switch to classes there is no warning.

**Comment by VoidAspect** [comment]
I have faced a similar problem. Here's a gist with a synthetic example:
https://gist.github.com/VoidAspect/8181c24b74f9c3f2c22e9730a98ac24e

**Comment by Zegveld** [comment]
Thanks for the report, the example project made it nice and easy to reproduce and to check if it is fixed in a newer version.
Tested it with the 1.5.0.Beta1 build, a problem still exists.
```
./mapstruct-bug/src/main/java/io/github/hejcz/DtoMapper.java:31: warning: Unmapped target property: "hes".
    WishList mapWishlist(WishList2 wishlist);
             ^
1 warning
```
Tested it with the upcoming 1.5.0.Beta2 build, problem is fixed.
```
./gradlew clean compileJava

BUILD SUCCESSFUL in 525ms
2 actionable tasks: 2 executed
```
Probably fixed by #2596 where a change was made in how we look at records.

Marking this one as closed:
It will be in the upcoming 1.5.0.Beta2 build and the 1.5.0 release

**Comment by KENNYSOFT** [comment]
@Zegveld Just for clarification: not #2596, but #2554
